### ISSUE TITLE
Blank Camera on Nexus 5

### ISSUE THREAD

**Issue by mrfleap** [tomthecarrot/arcore-for-all#20]
When building the APK and running it on my Nexus 5 through Android Studio, it says "Tango Core has stopped". When I tap outside of that box, the app seems to run fine however the box where the camera should be is blank. What's interesting is that there aren't any crashes in the logs (as far as I can tell).

System specs:
Device: Motorola Nexus 5
OS Version: 7.1.1
Stock Android

[Video (Sorry for Dropbox link)](https://www.dropbox.com/s/5ayd6t9g8iusouu/device-2017-08-31-201612.mp4?dl=0)

[Logcat](https://gist.github.com/mrfleap/07f4dca2af8916702dec8182e3ef3928)

By the way, thanks for sharing your work and having dedication to get it to work on more devices.

**Comment by khursani8** [comment]
@mrfleap Are you using http://tomthecarrot.com/projects/arcore-for-all/arcore-preview-modified.apk for arcore-preview? I stop getting "Tango Core has stopped" after using this apk

**Comment by tomthecarrot** [comment]
Thank you for your kind words and logging! I am noticing a pattern with some devices: the Tango 
 library refuses to check the Vulkan layer search path. I'm not too familiar with Vulkan, but it looks like an essential shared object isn't being loaded.

`ApplicationLoaders: ignored Vulkan layer search path /data/app/com.google.tango-2/lib/arm:/system/fake-libs:/data/app/com.google.tango-2/base.apk!/lib/armeabi-v7a for namespace 0xab52d090`

#7 #13 #18 have the same issue.

**Comment by tomthecarrot** [comment]
@khursani8 great news! Is ARCore working on your device now?

**Comment by mrfleap** [comment]
@khursani8 I was updating the post with the results as you commented, it seems it yields the same results on my device and still has the "Tango Core has stopped" prompt.

**Comment by khursani8** [comment]
@tomthecarrot no, just that the error "Tango Core has stopped" not pop up anymore

**Comment by tomthecarrot** [comment]
@khursani8 Interesting. Do you have a log? Which device are you using?

At least it doesn't crash anymore. That's a good start :P

**Comment by tomthecarrot** [comment]
@mrfleap Thanks for the info. Your device seems to be running without a certain shared object that it needs. It's evident in the other issues I referenced above. If we can find a pattern for this Vulkan search path issue, it likely will help isolate the issue so we can develop another modification.

**Comment by khursani8** [comment]
@tomthecarrot here is my log using Nexus 5
![photo_2017-09-01_11-56-46](https://user-images.githubusercontent.com/5404903/29954753-b67227a8-8f0c-11e7-8923-6b6c7ccf97ca.jpg)

[logcat.txt](https://github.com/tomthecarrot/arcore-for-all/files/1269120/logcat.txt)

**Comment by tomthecarrot** [comment]
@khursani8 Weird. I'm seeing the same type of Vulkan search path log in your device's output, but it doesn't crash the service like it does for @mrfleap:
`ApplicationLoaders: ignored Vulkan layer search path /mnt/asec/com.google.tango-1/lib/arm for namespace 0xb2a62090`

**Comment by mrfleap** [comment]
If it helps here is the "Phone status" screen on my device:

![device-2017-08-31-210242](https://user-images.githubusercontent.com/5022578/29954853-c2274b8c-8e8f-11e7-8a74-affb827f7a03.png)

**Comment by tomthecarrot** [comment]
@mrfleap Oh, I thought this was a Nexus 5. Can you confirm the issue title?

**Comment by mrfleap** [comment]
Oh dang, sorry I get these confused. This is a Nexus 6. Sorry.

**Comment by tomthecarrot** [comment]
@mrfleap No worries. Thanks for confirming. The differences in device behavior make more sense now ;)

**Comment by khursani8** [comment]
@tomthecarrot want me to create a new issue for Nexus 5?

**Comment by tomthecarrot** [comment]
@khursani8 That would be great. Thank you!

**Comment by Thaina** [comment]
I use Nexus 6 too and wish I can heard a good news sometimes

**Comment by mudassirzulfiqar** [comment]
got errors on nexus 6 

09-06 02:09:44.458 11754-11825/? I/Adreno: QUALCOMM build                   : a5b4970, If5818605d9
                                           Build Date                       : 10/12/16
                                           OpenGL ES Shader Compiler Version: XE031.09.00.04
                                           Local Branch                     : N24D
                                           Remote Branch                    : 
                                           Remote Branch                    : 
                                           Reconstruct Branch               : 
09-06 02:09:44.477 11754-11825/? I/OpenGLRenderer: Initialized EGL, version 1.4
09-06 02:09:44.477 11754-11825/? D/OpenGLRenderer: Swap behavior 1
09-06 02:09:46.049 11754-11761/? I/art: Ignoring second debugger -- accepting and dropping
09-06 02:09:47.354 11754-11754/? E/SupportedDevices: Device calibration unavailable. Just kidding! ;)
09-06 02:09:47.364 11754-11754/? I/TangoClientLibLoader: Pure Java path, not loading libtango_client_api.so at all.
09-06 02:09:47.375 11754-11754/? D/ApplicationLoaders: ignored Vulkan layer search path /data/app/com.google.tango-2/lib/arm:/system/fake-libs:/data/app/com.google.tango-2/base.apk!/lib/armeabi-v7a for namespace 0x9e45a090
09-06 02:09:47.387 11754-11754/? I/ndk_camera: Starting native camera system initialization.
09-06 02:09:47.389 11754-11754/? D/ndk_camera: Camera 0 has 82 metadata tags
09-06 02:09:47.389 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 1310721
09-06 02:09:47.389 11754-11754/? W/ndk_camera: Failed to get metadata tag 1310721 for camera 0 with error code -10004
09-06 02:09:47.389 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 851977
09-06 02:09:47.389 11754-11754/? W/ndk_camera: Failed to get metadata tag 851977 for camera 0 with error code -10004
09-06 02:09:47.389 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 1638401
09-06 02:09:47.389 11754-11754/? W/ndk_camera: Failed to get metadata tag 1638401 for camera 0 with error code -10004
09-06 02:09:47.390 11754-11754/? D/ndk_camera: Camera 1 has 81 metadata tags
09-06 02:09:47.390 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 1310721
09-06 02:09:47.390 11754-11754/? W/ndk_camera: Failed to get metadata tag 1310721 for camera 1 with error code -10004
09-06 02:09:47.390 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 851977
09-06 02:09:47.390 11754-11754/? W/ndk_camera: Failed to get metadata tag 851977 for camera 1 with error code -10004
09-06 02:09:47.390 11754-11754/? E/ACameraMetadata: getConstEntry: cannot find metadata tag 1638401
09-06 02:09:47.390 11754-11754/? W/ndk_camera: Failed to get metadata tag 1638401 for camera 1 with error code -10004
09-06 02:09:47.390 11754-11754/? I/ndk_camera: Camera system initialized successfully with 2 cameras.
09-06 02:09:47.390 11754-11754/? W/tango_camera_native_jni: Camera system initilized.
09-06 02:09:47.390 11754-11754/? E/tango_camera_native: 2 is not a valid camera index.
09-06 02:09:47.390 11754-11754/? E/tango_camera_native: 3 is not a valid camera index.
09-06 02:09:47.395 11754-11754/? I/tango_camera_native_jni: Attaching texture_window 0xa41b3808
09-06 02:09:47.396 11754-11754/? D/TangoVhs: about to bind as vhs
09-06 02:09:47.409 11754-11754/? D/TangoVhs: finished bind as vhs
09-06 02:09:48.278 11754-11754/? I/Choreographer: Skipped 50 frames!  The application may be doing too much work on its main thread.
